test/test_retry_deprecated.py

```diff
@@ -0,0 +1,428 @@
+import mock
```

Comment:
This is literally a copy-paste of `test_retry.py` with extra asserts about deprecated options. Designed so that we can move along with `test_retry.py` without having to think about deprecations much but also ensure that everything that used to work still does.